# HG changeset patch
# User daybreaker
# Date 1306421524 -32400
# Branch py3k-workaround
# Node ID 0067e50fad8a16e68ac20f594f1a7dbbe13e0cf1
# Parent  47a94f723e803489e0608eeb39a89494b214c068
Made it working with Python 3.2 on Windows 7 Korean version + UTF-8 encoded source files.
Another workaround is required due to `next()` -> `__next__()`, see https://bitbucket.org/birkenfeld/sphinx/issue/635/no-attribute-next

diff -r 47a94f723e80 -r 0067e50fad8a sphinx/pycode/__init__.py
--- a/sphinx/pycode/__init__.py	Sun May 15 14:54:52 2011 +0200
+++ b/sphinx/pycode/__init__.py	Thu May 26 23:52:04 2011 +0900
@@ -179,7 +179,7 @@
         if ('file', filename) in cls.cache:
             return cls.cache['file', filename]
         try:
-            fileobj = open(filename, 'r')
+            fileobj = open(filename, 'r', encoding='utf8')
         except Exception, err:
             raise PycodeError('error opening %r' % filename, err)
         obj = cls(fileobj, modname, filename)
diff -r 47a94f723e80 -r 0067e50fad8a sphinx/util/__init__.py
--- a/sphinx/util/__init__.py	Sun May 15 14:54:52 2011 +0200
+++ b/sphinx/util/__init__.py	Thu May 26 23:52:04 2011 +0900
@@ -241,7 +241,8 @@
 
     def find_cookie(line):
         try:
-            line_string = line.decode('ascii')
+            #line_string = line.decode('ascii')
+            line_string = line
         except UnicodeDecodeError:
             return None
 
@@ -252,9 +253,10 @@
 
     default = sys.getdefaultencoding()
     first = read_or_stop()
-    if first and first.startswith(BOM_UTF8):
-        first = first[3:]
-        default = 'utf-8-sig'
+    # The file is read using a specific encoding already.
+    #if first and first.startswith(BOM_UTF8):
+    #    first = first[3:]
+    #    default = 'utf-8-sig'
     if not first:
         return default
     encoding = find_cookie(first)

